rpc/gnmi/gnmi-union_replace.md

```diff
@@ -0,0 +1,327 @@
+  - [3. Problem Statement](#3-problem-statement)
+    - [3.1 The need for mixed origin configuration](#31-the-need-for-mixed-origin-configuration)
+    - [3.2 What origins are to be mixed?](#32-what-origins-are-to-be-mixed)
+  - [4. Device behavior for single-origin SetRequests](#4-device-behavior-for-single-origin-setrequests)
+    - [4.1 Replace operation for Origin CLI  behavior](#41-replace-operation-for-origin-cli--behavior)
+    - [4.2 Replace operation for path based Native Origin behavior](#42-replace-operation-for-path-based-native-origin-behavior)
+  - [5 Device behavior for mixed native/CLI OC origin  SetRequests](#5-device-behavior-for-mixed-nativecli-oc-origin--setrequests)
+    - [5.1 Objective](#51-objective)
+    - [5.2 Requirements for SetRequest with 2 origins](#52-requirements-for-setrequest-with-2-origins)
+      - [5.2.2 union\_replace operation containing a native origin and origin OpenConfig](#522-union_replace-operation-containing-a-native-origin-and-origin-openconfig)
+    - [5.3 Union behavior options](#53-union-behavior-options)
+      - [5.3.1 union behavior for CLI and OpenConfig](#531-union-behavior-for-cli-and-openconfig)
+      - [5.3.2 union behavior for Native and OpenConfig](#532-union-behavior-for-native-and-openconfig)
+      - [5.3.3 Resolving issues with union between the origins](#533-resolving-issues-with-union-between-the-origins)
+        - [5.3.3.1 Default values](#5331-default-values)
+        - [5.3.3.2 Overlapping values in CLI and OC](#5332-overlapping-values-in-cli-and-oc)
+        - [Option 1: Resolving issues with union of CLI and OC values with error](#option-1-resolving-issues-with-union-of-cli-and-oc-values-with-error)
+        - [Option 2: Resolving issues with merging CLI and OC by overwriting](#option-2-resolving-issues-with-merging-cli-and-oc-by-overwriting)
+      - [5.3.6 Overlapping values in NY and OC](#536-overlapping-values-in-ny-and-oc)
+      - [5.3.7 Example of resolving default values](#537-example-of-resolving-default-values)
+        - [Scenario A](#scenario-a)
+        - [Scenario B](#scenario-b)
+  - [6 Examples](#6-examples)
+    - [6.1 Examples of expected configuration work flows](#61-examples-of-expected-configuration-work-flows)
+    - [6.3 union\_replace example](#63-union_replace-example)
+  - [7. gNSI, bootz and gNMI interoperation](#7-gnsi-bootz-and-gnmi-interoperation)
+  - [8. References](#8-references)
+
+## 1. Objective
+
+There is an operational goal to ensure a network operating system’s (NOS) complete operating configuration state is matched to the intended configuration state.  A simple way to guarantee this is by generating the entire configuration for the device and replacing whatever configuration the device currently has.  This declarative specification of the configuration is colloquially referred to as ‘full config replace’.
+Although the concept is simple enough, there are portions of the configuration which may not be modeled in OpenConfig.
```

Comment:
Suggest to add something here that states something like, "gNMI already supports this concept through the definition of multiple schemas, see https://github.com/openconfig/reference/blob/master/rpc/gnmi/mixed-schema.md".

We should consider what our approach for this document's coexistence with the previous document is -- the `origin` field's meaning is established in the mixed schema document. Maybe removing the example from mixed schema (or making it simpler to include non-overlapping information) and then referencing this document for more complex cases is something that would ensure usability for documentation readers.

Reply:
I will move the detailed description of 'origin' from the mixed-schema doc to the gnmi specification.  I think the mixed schema doc then would be left only with things that overlap with union_replace and could be deleted.  I'll send another commit with the changes for your view